vispy/visuals/gridlines.py

```diff
@@ -57,6 +64,20 @@
     if (alpha == 0.) {
         discard;
     }
+
+    // outside the defined region
+    if (any(lessThan(local_pos.xy, u_gridlines_bounds.xz)) ||
+        any(greaterThan(local_pos.xy, u_gridlines_bounds.yw))) {
+
+        // inside the defined border width
+        if (all(greaterThan(local_pos.xy, u_gridlines_bounds.xz - u_border_width)) &&
+            all(lessThan(local_pos.xy, u_gridlines_bounds.yw + u_border_width))) {
```

Comment:
If I understand correctly, this adds a border *outside* of the grid. I feel like maybe it's more exptected to be "on top of the outer line", with half the thickness inside and half outside... Not a big deal, just pointing out in case it's preferred.

Reply:
If it's half-half, there will be strange corner cases.

For example, say, the grid is 1x1 but the thickness is 5. Then, the whole "grid" would be the boarder...

Reply:
I think it is pretty standard/common for it to be half-half. For example, I just made a rectangle on a Google Drawing (google drive) and increasing the border size keeps it on the line of the rectangle, but it "grows" toward the inside and the outside (half on the inside, half on the outside).